config/routes.rb

```diff
@@ -55,7 +59,12 @@
     end
   end
 
-  resources :events, only: [:index, :show]
+  resources :events,         only: [:index, :show]
+  resources :ongoing_events, only: [:index, :show] do
+    post   :opt_in,  on: :member
```

Comment:
Would this be a candidate for a concern?
```ruby
concern :participation do
  post :opt_in
  delete :opt_out
end

....

resources :ongoing_events, only: [:index, :show], concerns: :participation
resources :shifts, only: [:show], concerns: :participation
```

Just came to my mind, rare occasion where this could be useful.